packages/workspace/src/Workspace.ts

```diff
@@ -0,0 +1,105 @@
+import { Api } from '@scalecube/scalecube-microservice';
+
+import { Workspace as IWorkspace } from './api/Workspace';
+import { ServicesMap, ServicesRequest } from './api/methods/services';
+import { ComponentsMap, ComponentsRequest, Component } from './api/methods/components';
+import WorkspaceConfig from './api/WorkspaceConfig';
+import { RegisteredService, RegisterServiceRequest } from './api/methods/registerService';
+import { ComponentRegistry, ServiceRegistry } from './helpers/types';
+import {
+  componentToRegisterMissingInConfigurationError,
+  invalidRegisterServiceRequestError,
+  serviceAlreadyRegisteredError,
+  serviceToRegisterMissingInConfigurationError,
+} from './helpers/const';
+import {
+  validateComponentInConfig,
+  validateRegisterServiceRequest,
+  validateServiceInConfig,
+} from './helpers/validators';
+
+export class Workspace implements IWorkspace {
+  private configuration: WorkspaceConfig;
+  private serviceRegistry: ServiceRegistry;
+  private componentRegistry: ComponentRegistry;
+  private microservice?: Api.Microservice;
+
+  constructor(configuration: WorkspaceConfig, init: (arg: any) => Promise<any>) {
+    this.configuration = configuration;
+    this.serviceRegistry = {} as ServiceRegistry;
+    this.componentRegistry = {} as ComponentRegistry;
+
+    init(this)
+      .then((data) => {
+        this.microservice = data.microservice;
+      })
+      .catch(() => {});
+  }
+
+  services(servicesRequest: ServicesRequest): Promise<ServicesMap> {
+    const services = Object.values(this.serviceRegistry);
+
+    return Promise.resolve(
+      services.reduce(
+        (servicesMap: ServicesMap, service: RegisteredService) => {
+          return {
+            ...servicesMap,
+            [service.serviceName]: Promise.resolve({
```

Comment:
What happens here in case some services didn't called yet `registerService(...)` ?

Reply:
We iterate over registered services, so services, that haven't been registered yet, are not included

Reply:
We didn't plan to include them from the beginning and having their promise resolved only when they will be registered ?